File: app/src/main/java/com/example/khughes/machewidget/VehicleIds.java
package com.example.khughes.machewidget;

import java.util.Objects;

public class VehicleIds {
    private String VIN;
    private String nickname;
    private boolean enabled;

    public String getVIN() {
        return VIN;
    }

    public void setVIN(String VIN) {
        this.VIN = VIN;
    }

    public String getNickname() {
        return nickname;
    }

    public void setNickname(String nickname) {
        this.nickname = nickname;
    }

    public boolean isEnabled() {
        return enabled;
    }

    public void setEnabled(boolean enabled) {
        this.enabled = enabled;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        VehicleIds that = (VehicleIds) o;
        return Objects.equals(VIN, that.VIN) && Objects.equals(nickname, that.nickname);
    }

    @Override
    public int hashCode() {
        return Objects.hash(VIN, nickname);
    }
}

File: app/src/main/java/com/example/khughes/machewidget/VehicleIds.kt
package com.example.khughes.machewidget

import androidx.room.ColumnInfo
import java.util.*

class VehicleIds {
    @ColumnInfo(name = "car_vehicleId")
    var vehicleId: String = ""
    @ColumnInfo(name = "car_nickName")
    var nickname: String? = null
    var modelId: Vehicle.Companion.Model = Vehicle.Companion.Model.UNKNOWN
    var enabled = false
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val that = other as VehicleIds
        return vehicleId == that.vehicleId && nickname == that.nickname
    }

    override fun hashCode(): Int {
        return Objects.hash(vehicleId, nickname)
    }
}